### ISSUE TITLE
tf.data.Dataset.from_tensor_slices creates infinite loop in https://www.tensorflow.org/tutorials/load_data/pandas_dataframe 

### ISSUE THREAD

**Issue by cmp1** [tensorflow/tensorflow#33631]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- TensorFlow tutorial 
- TensorFlow installed from: Source
- TensorFlow version: 1.14
- Python version: 3.7.3

**Describe the current behaviour**

Attempting to create a batched dataset to train an RNN-LSTM using the logic from the [TensorFlow Time Series Forecasting tutorial ](https://www.tensorflow.org/tutorials/structured_data/time_series) and the [load Pandas.DataFrame logic](https://www.tensorflow.org/tutorials/load_data/pandas_dataframe).

When I execute these lines on my Pandas Dataframe of shape (1661, 5):

```
dataset = tf.data.Dataset.from_tensor_slices((df.values, target.values))
for feat, targ in dataset.take(5):
  print ('Features: {}, Target: {}'.format(feat, targ))

```
The loop runs as an infinite loop and does not produce the same out as that in the tutorial. The only difference with my dataframe is that it has a DateTimeIndex.

The output 
**Describe the expected behaviour**

I would expect a batch of size 5 as per dataset.take(5) in the format below:

Features: [ 63.    1.    1.  145.  233.    1.    2.  150.    0.    2.3   3.    0.
   2. ], Target: 0

**Code to reproduce the issue**
`df.head(10)`

Out:[1]
	index	col_1         col_2        col_3       col_4        col_5
2016-10-03	0.017647	0.000000	0.048040	0.334455	0.132643
2016-10-04	0.017647	0.000925	0.047372	0.334455	0.174480
2016-10-05	0.017647	0.001849	0.046696	0.334455	0.142559
2016-10-06	0.017647	0.002774	0.046011	0.334455	0.148387
2016-10-07	0.017647	0.003699	0.045317	0.334455	0.142472
2016-10-10	0.017647	0.004624	0.044615	0.333301	0.129686
2016-10-11	0.017647	0.005548	0.043905	0.331875	0.120031
2016-10-12	0.017647	0.006473	0.043185	0.324570	0.120031
2016-10-13	0.017647	0.007398	0.042458	0.324327	0.105506
2016-10-14	0.017647	0.008323	0.041721	0.341778	0.129425

`
target = final_df.pop('col_1')
dataset = tf.data.Dataset.from_tensor_slices((final_df.values, final_df.values))
for feat, targ in dataset.take(5):
  print ('Features: {}, Target: {}'.format(feat, targ))
`
Out[2]:

Features: Tensor("IteratorGetNext:0", shape=(4,), dtype=float64), Target: Tensor("IteratorGetNext:1", shape=(4,), dtype=float64)
Features: Tensor("IteratorGetNext_1:0", shape=(4,), dtype=float64), Target: Tensor("IteratorGetNext_1:1", shape=(4,), dtype=float64)
target: Tensor("IteratorGetNext_9:1", shape=(4,), dtype=float64)

Output in an infinite loop.

**Comment by ravikyram** [comment]
@cmp1 

Can you please share sample data and simple standalone code to reproduce the issue in our environment.It will help us localizing the issue faster.Thanks!

**Comment by cmp1** [comment]
[df.xlsx](https://github.com/tensorflow/tensorflow/files/3769609/df.xlsx)

Hi - thank you for getting back to me. I have attached a sample from the dataframe.  The code I followed is exactly that from the RNN tutorial. Previously I have used generator functions to train RNNs however this function looks like it could substantially reduce the time creating these:

`
# separate label from df
target = df.pop('label')

dataset = tf.data.Dataset.from_tensor_slices((df.values, target.values))

# loops over entire dataset and does not produce tensors of size n features, target 
for feat, targ in dataset.take(5):
  print ('Features: {}, Target: {}'.format(feat, targ))

# runs ok
train_dataset = dataset.shuffle(len(df)).batch(1)

# define model
def get_compiled_model():
  model = tf.keras.Sequential([
    tf.keras.layers.Dense(10, activation='relu'),
    tf.keras.layers.Dense(10, activation='relu'),
    tf.keras.layers.Dense(1, activation='sigmoid')
  ])

  model.compile(optimizer='adam',
                loss='mean_squared_error',
                metrics=['mae', 'acc'])
  return model

model = get_compiled_mo

model = get_compiled_model()

# returns error
model.fit(train_dataset, epochs=15)
`

Thank you!

**Comment by ravikyram** [comment]
@cmp1 
Can you please enable the eager execution by `tf. enable_eager_execution()`at the start of the code and check. After enabling eager execution i am not seeing infinite loop in the output. Thanks!

**Comment by cmp1** [comment]
Ah I forgot this - that has fixed it! thank you so much!

**Comment by ravikyram** [comment]
I am closing this issue since the query is been resolved.Thanks!
